Source/Server/Managers/DiscordManager.cs

```diff
@@ -0,0 +1,154 @@
+using System.Text;
+using Discord;
+using Discord.WebSocket;
+
+namespace GameServer
+{
+    public static class DiscordManager
+    {
+        private static DiscordSocketClient? _client;
+        private static Queue<string> _consoleBuffer = new();
+        private static Task sendToConsoleChannel = new(SendMessageToConsoleChannel);
+        private static Task updatePlayerCount = new(UpdatePlayerCount);
+        private static int _sendToConsoleDelay = 1000;
+        private static int _updatePlayerCountDelay = 60000;
+
+        private static DiscordSocketConfig Config = new() { GatewayIntents = GatewayIntents.Guilds | GatewayIntents.GuildMessages | GatewayIntents.MessageContent };
+
+        public static async void StartDiscordIntegration()
```

Comment:
Add validation checks for _Master.serverConfig_ to help ensure the required fields of _DiscordIntegration.Token_ is set up properly before starting anything Discord-related.

public static async void StartDiscordIntegration()
{
    if (Master.serverConfig == null || string.IsNullOrEmpty(Master.serverConfig.DiscordIntegration.Token))
    {
        Logger.Error("Discord token is not configured.");
        return;
    }

    _client = new DiscordSocketClient(Config);

    // existing logic...
}

Reply:
Quick realization. Instead of using async void async task would be the better option because it allows callers to await these methods and handle any exception much better.